SecretSanta/test/SecretSanta.Api.Tests/Controllers/UsersControllerTests.cs

```diff
@@ -0,0 +1,169 @@
+
+        [TestMethod]
+        public void Get_GivenInvalidId_ReturnsNotFound()
+        {
+            mockRepo mock = new();
+            mock.GetItemUser = null!;
+            UsersController sut = new(mock);
+
+            ActionResult<User> result = sut.Get(0);
+
+            Assert.IsTrue(result.Result is NotFoundResult);
+        }
+
+        [TestMethod]
+        public void Delete_GivenValidId_ReturnsOk()
+        {
+            mockRepo mock = new();
+            mock.RemoveSuccess = true;
+            UsersController sut = new(mock);
+
+            ActionResult result = sut.Delete(0);
+
+            Assert.IsTrue(result is OkResult);
+        }
+
+        [TestMethod]
+        public void Delete_GivenInvalidId_ReturnsNotFound()
+        {
+            mockRepo mock = new();
+            mock.RemoveSuccess = false;
+            UsersController sut = new(mock);
+
+            ActionResult result = sut.Delete(0);
+
+            Assert.IsTrue(result is NotFoundResult);
+        }
+
+        [TestMethod]
+        public void Post_GivenValidUser_ReturnsUser()
+        {
+            User expectedUser = new User() { Id = 0, FirstName = "test", LastName = "ing" };
+            mockRepo mock = new();
+            UsersController sut = new(mock);
+
+            ActionResult<User?> result = sut.Post(expectedUser);
+
+            Assert.AreEqual(expectedUser, result.Value);
+        }
+
+        [TestMethod]
+        public void Post_GivenInvalidUser_ReturnsBadRequest()
+        {
+            UsersController sut = new(new mockRepo());
+
+            ActionResult<User?> result = sut.Post(null!);
+
+            Assert.IsTrue(result.Result is BadRequestResult);
+        }
+
+        [TestMethod]
+        public void Put_GivenValidData_UpdatesUser()
+        {
+            DtoUser update = new() { FirstName = "up", LastName = "date" };
+            User olduser = new() { Id = 0, FirstName = "test", LastName = "ing" };
+            mockRepo mock = new();
+            mock.GetItemUser = olduser;
+            UsersController sut = new(mock);
+
+            ActionResult result = sut.Put(0, update);
+
+            Assert.AreEqual(update.FirstName, olduser.FirstName);
+            Assert.AreEqual(update.LastName, olduser.LastName);
+            Assert.IsTrue(result is OkResult);
+        }
+
+        [TestMethod]
+        public void Put_GivenInvalidDto_ReturnsBadRequest()
+        {
+            UsersController sut = new(new mockRepo());
+
+            ActionResult result = sut.Put(0, null!);
+
+            Assert.IsTrue(result is BadRequestResult);
+        }
+
+        [TestMethod]
+        public void Put_GivenInvalidId_ReturnsNotFound()
+        {
+            mockRepo mock = new();
+            mock.GetItemUser = null!;
+            UsersController sut = new(mock);
+
+            ActionResult result = sut.Put(0, new DtoUser());
+
+            Assert.IsTrue(result is NotFoundResult);
+        }
+
+
+        private class mockRepo : IUserRepository

```

Comment:
Nit: Classes should be named with PascalCase even when they are private.
```suggestion
        private class MockRepo : IUserRepository

```